crates/proto/src/tcp/tcp_client_stream.rs

```diff
@@ -138,125 +140,34 @@ impl Connect for TokioTcpStream {
-                }
-            }
-
-            panic!("timeout");
-        })
-        .unwrap();
-
-    // TODO: need a timeout on listen
-    let server = std::net::TcpListener::bind(SocketAddr::new(server_addr, 0)).unwrap();
-    let server_addr = server.local_addr().unwrap();
-
-    let send_recv_times = 4;
-
-    // an in and out server
-    let server_handle = std::thread::Builder::new()
-        .name("test_tcp_client_stream_ipv4:server".to_string())
-        .spawn(move || {
-            let (mut socket, _) = server.accept().expect("accept failed");
-
-            socket
-                .set_read_timeout(Some(std::time::Duration::from_secs(5)))
-                .unwrap(); // should receive something within 5 seconds...
-            socket
-                .set_write_timeout(Some(std::time::Duration::from_secs(5)))
-                .unwrap(); // should receive something within 5 seconds...
-
-            for _ in 0..send_recv_times {
-                // wait for some bytes...
-                let mut len_bytes = [0_u8; 2];
-                socket
-                    .read_exact(&mut len_bytes)
-                    .expect("SERVER: receive failed");
-                let length =
-                    u16::from(len_bytes[0]) << 8 & 0xFF00 | u16::from(len_bytes[1]) & 0x00FF;
-                assert_eq!(length as usize, TEST_BYTES_LEN);
-
-                let mut buffer = [0_u8; TEST_BYTES_LEN];
-                socket.read_exact(&mut buffer).unwrap();
-
-                // println!("read bytes iter: {}", i);
-                assert_eq!(&buffer, TEST_BYTES);
-
-                // bounce them right back...
-                socket
-                    .write_all(&len_bytes)
-                    .expect("SERVER: send length failed");
-                socket
-                    .write_all(&buffer)
-                    .expect("SERVER: send buffer failed");
-                // println!("wrote bytes iter: {}", i);
-                std::thread::yield_now();
-            }
-        })
-        .unwrap();
-
-    // setup the client, which is going to run on the testing thread...
-    let mut io_loop = runtime::Runtime::new().unwrap();
-
-    // the tests should run within 5 seconds... right?
-    // TODO: add timeout here, so that test never hangs...
-    // let timeout = Timeout::new(Duration::from_secs(5));
-    let (stream, mut sender) = TcpClientStream::<TokioTcpStream>::new(server_addr);
-
-    let mut stream = io_loop.block_on(stream).expect("run failed to get stream");
-
-    for _ in 0..send_recv_times {
-        // test once
-        sender
-            .send(SerialMessage::new(TEST_BYTES.to_vec(), server_addr))
-            .expect("send failed");
-        let (buffer, stream_tmp) = io_loop.block_on(stream.into_future());
-        stream = stream_tmp;
-        let buffer = buffer
-            .expect("no buffer received")
-            .expect("error recieving buffer");
-        assert_eq!(buffer.bytes(), TEST_BYTES);
+mod tests {
```

Comment:
No additional value. I thought it would follow the pattern of other crates, which means the unit tests are defined in the same file as where the function are defined. Either way works. I can move them to the tests module.

Reply:
Yeah, I don't know... I mean, running all the tests on proto is enough to execute the coverage for this stuff... so it should be good.

We could consider a generic executor and fake TCP sockets at some point in the future to test the code in this module if we want... and those could live here.

Reply:
So I take it as I don't need to move it.

Reply:
you can leave them if you think it keeps things in a state that makes it less likely we'll break alternative implementations. So no need to move.

Reply:
It seems I replied to all the comments. I also pushed a newer version.
Regarding the comments about where the tests could be, when you mentioned integration test, you thought about `trust-dns/tests/integration-tests`? I am not expecting a breakage now at least as long as the trait def of TcpStream, UdpSocket and Executor stay. Probably we can rethink about integration test on top of resolver crate when I am able to import this crate into Fuchsia? 

BTW, I put some comments for pull #935 last night and just realized I did not submit it this morning.